### ISSUE TITLE
Sub 1% chance breaks the plugin

### ISSUE THREAD

**Issue by Folas1337** [zizmax/AcuteLoot#41]
```yaml
rarities:
  1000:
    name: "5"
    color: "&5"
    chance: 0.66
    effect-chance: 0
```
^ This configuration breaks the plugin as while it's loading it complains:
```
[12:28:36 INFO]: [AcuteLoot] Enabling AcuteLoot v2.13.0
[12:28:36 INFO]: [AcuteLoot] +----------------------------------------------------------------+
[12:28:36 INFO]: [AcuteLoot] |                      AcuteLoot Community                       |
[12:28:36 INFO]: [AcuteLoot] +================================================================+
[12:28:36 INFO]: [AcuteLoot] | * Please report bugs at: https://git.io/JkJLD                  |
[12:28:36 INFO]: [AcuteLoot] | * Join the AcuteLoot Discord at: https://discord.gg/BXhUUQEymg |
[12:28:36 INFO]: [AcuteLoot] | * Enjoying the plugin? Leave a review and share with a friend! |
[12:28:36 INFO]: [AcuteLoot] +----------------------------------------------------------------+
[12:28:36 INFO]: [AcuteLoot] Disabling modules [NONE]
[12:28:36 INFO]: [AcuteLoot] Pre-Enabling modules lootRules
[12:28:36 INFO]: [AcuteLoot] Initialized 208 materials
[12:28:36 INFO]: [AcuteLoot] Loaded 5 name generators
[12:28:36 ERROR]: Error occurred while enabling AcuteLoot v2.13.0 (Is it up to date?)
java.lang.IllegalArgumentException: Relative chance must be positive.
        at acute.loot.acutelib.util.Checks.check(Checks.java:105) ~[AcuteLoot.jar:?]
        at acute.loot.acutelib.util.Checks.check(Checks.java:86) ~[AcuteLoot.jar:?]
        at acute.loot.acutelib.util.Checks.requirePositive(Checks.java:47) ~[AcuteLoot.jar:?]
        at acute.loot.acutelib.collections.IntegerChancePool.add(IntegerChancePool.java:82) ~[AcuteLoot.jar:?]
        at acute.loot.AcuteLoot.reloadConfiguration(AcuteLoot.java:344) ~[AcuteLoot.jar:?]
        at acute.loot.AcuteLoot.onEnable(AcuteLoot.java:127) ~[AcuteLoot.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[paper-api-1.19-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[paper-api-1.19-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:542) ~[paper-api-1.19-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_19_R1.CraftServer.enablePlugin(CraftServer.java:563) ~[paper-1.19.jar:git-Paper-77]
        at org.bukkit.craftbukkit.v1_19_R1.CraftServer.enablePlugins(CraftServer.java:477) ~[paper-1.19.jar:git-Paper-77]
        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:634) ~[paper-1.19.jar:git-Paper-77]
        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:420) ~[paper-1.19.jar:git-Paper-77]
        at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:306) ~[paper-1.19.jar:git-Paper-77]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1122) ~[paper-1.19.jar:git-Paper-77]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:303) ~[paper-1.19.jar:git-Paper-77]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
[12:28:36 INFO]: [AcuteLoot] Disabling AcuteLoot v2.13.0
[12:28:36 INFO]: [AcuteLoot] Disabled
```
I am pretty sure anything below 1% is still positive but for some reason the plugin doesn't think so. I hope that's not by design or for some technical reason but just an oversight in some way.

Thanks in advance :)

**Comment by zizmax** [comment]
Thanks for reporting this behavior. Turns out, this is actually working as expected. The system AcuteLoot uses for chances isn't super intuitive. We created an "integer chance pool" to handle rarities to make it easier for people to create custom rarities without having to always make sure everything explicitly adds up to 100%. I've added a fair bit of explanation to the [config.yml](https://github.com/zizmax/AcuteLoot/commit/08b29f7d6ad95ebd2d0842a35b1f76828d5da923) to hopefully make it clearer how it works and how to configure chances that have probabilities < 1%.
